File: ruby/observer/NewsLetter/observer/subscriber.rb
require './observer/observer'
require './subject/newsletter'

class Subscriber < Observer

  def initialize(newsletter)
    @name = ''
    @newsletter = Newsletter.new
    @newsletter = newsletter
    @newsletter.register_observer(self)
  end

  def update(name)
    super
    @name = name
    puts("New Subscriber: +" + @name + " added to the newsletter")
  end

end

File: cpp/observer/NewsLetter/observer/Subscriber.cpp
#include <iostream>

using namespace std;

class Subscriber : Observer {

public:
    explicit Subscriber(Newsletter* newsletter) {
        this->newsletter = newsletter;
        this->newsletter->registerObserver(this);
    }

    void update(string name) override {
        _name = name;
        cout << "New Subscriber: +" << _name << " added to the newsletter" << endl;
    }

private:
    Newsletter* newsletter = new Newsletter();
    string _name;
};
